src/Task.ts

```diff
@@ -422,73 +271,11 @@ export class Task {
-            taskString += this.componentToString(taskLayout, component);
-        }
-        return taskString;
-    }
-
-    /**
-     * Renders a specific TaskLayoutComponent of the task (its description, priority, etc) as a string.
-     */
-    public componentToString(layout: TaskLayout, component: TaskLayoutComponent) {
-        switch (component) {
-            case 'description':
-                return this.description;
-            case 'priority': {
-                let priority: string = '';
-
-                if (this.priority === Priority.High) {
-                    priority = ' ' + prioritySymbols.High;
-                } else if (this.priority === Priority.Medium) {
-                    priority = ' ' + prioritySymbols.Medium;
-                } else if (this.priority === Priority.Low) {
-                    priority = ' ' + prioritySymbols.Low;
-                }
-                return priority;
-            }
-            case 'startDate':
-                if (!this.startDate) return '';
-                return layout.options.shortMode
-                    ? ' ' + startDateSymbol
-                    : ` ${startDateSymbol} ${this.startDate.format(TaskRegularExpressions.dateFormat)}`;
-            case 'createdDate':
-                if (!this.createdDate) return '';
-                return layout.options.shortMode
-                    ? ' ' + createdDateSymbol
-                    : ` ${createdDateSymbol} ${this.createdDate.format(TaskRegularExpressions.dateFormat)}`;
-            case 'scheduledDate':
-                if (!this.scheduledDate || this.scheduledDateIsInferred) return '';
-                return layout.options.shortMode
-                    ? ' ' + scheduledDateSymbol
-                    : ` ${scheduledDateSymbol} ${this.scheduledDate.format(TaskRegularExpressions.dateFormat)}`;
-            case 'doneDate':
-                if (!this.doneDate) return '';
-                return layout.options.shortMode
-                    ? ' ' + doneDateSymbol
-                    : ` ${doneDateSymbol} ${this.doneDate.format(TaskRegularExpressions.dateFormat)}`;
-            case 'dueDate':
-                if (!this.dueDate) return '';
-                return layout.options.shortMode
-                    ? ' ' + dueDateSymbol
-                    : ` ${dueDateSymbol} ${this.dueDate.format(TaskRegularExpressions.dateFormat)}`;
-            case 'recurrenceRule':
-                if (!this.recurrence) return '';
-                return layout.options.shortMode
-                    ? ' ' + recurrenceSymbol
-                    : ` ${recurrenceSymbol} ${this.recurrence.toText()}`;
-            case 'blockLink':
-                return this.blockLink ?? '';
-            default:
-                throw new Error(`Don't know how to render task component of type '${component}'`);
-        }
+    public toString(): string {
+        return getTaskFormat().taskSerializer.serialize(this);
```

Comment:
I think it would be worth documenting that this will be formatted in the user's chosen format.

I tried to work out if all of the following used the format from user settings, but ran out of time:

- toLi
- toString
- toFileLineString

I am not saying there is a problem here - I honestly don't know if there is - but I think it would be helpful for the formatting behaviour of each to be documented.